roadmap/implementers-guide/src/runtime/scheduler.md

```diff
@@ -1,3 +1,227 @@
+
+#### Runtime Implementation
+
+More details on auctioning
+[here](https://www.notion.so/paritytechnologies/Exotic-Scheduling-Auctions-53148e8d90d74411ab445b7d1dd6cfc8).
+
+- Bid has fee on top of bid for spam protection - fee can be quite low, it should
+  not harm the sender much, if losing a bid and a resend is necessary. Just high
+  enough so spam protection is maintained.
+- Minimum bid - configurable.
+- Lifetimes
+
+#### Possible Optimizations
+
+With the above, a user has to place an individual bid for each and every core
+desired. This is quite wasteful, if multiple cores are needed as a point in
+time. As each of these bids would likely be identical and each of them would
+need to be signed. What we can do is extend the format of the extrinsic slightly
+to contain a number specifying how many cores are desired and whether parallel
+cores are allowed (winning two cores in a single block). Likely a `max_parallel`
+parameter. Then the user can submit a single extrinsic, with only a single
+signature biding for e.g. 10 cores. Internally we treat those as indidivual
+equal bids (all the same price), competing individually with all the other bids.
+Therefore it is possible that only a fraction of the desired cores will actually
+be won. E.g. just one or two out of the ten desired.
+
+
+## Where is the money going?
+
+TODO: Bids and fees - where are they spent, what pallet do we need to interact here
+with?
+
+## Scheduler/Collators/Validators
+
+1. Collators need to know in advance when they should connect to which validator
+   group to prepare and send a collation.
+2. Collators need to know in advance what collators for which
+   parachains/parathreads they should accept connections and collations from.
+3. Users need to know whether they won a slot or not - seeing assigned cores
+   could be one way of achieving that goal. (Likely not very suitable though).
+
+### Interface to the scheduler
+
+Once we found claims for cores, we need to get them to the scheduler. This
+interface will roughly be:
+
+Give me claims for this claim queue size for this number of cores. For
+simplicity we make the number of cores per assignment provider a configuration
+that can only be changed on session boundaries.
+
+Basically the scheduler needs to be able to provide core assignments for the
+look ahead period required by asynchronous backing. The assignment providers
+(like the parathread one above) need to be able to provide those, given a
+configured number of available cores to the provider.
+
+Assuming we treat cores separately and sequentially (see questions below), an
+assignment provider could return an enum like this:
+
+```rust
+enum Assignment {
+  // Parathreads
+  List(Vec<Claim>),
+  // slot sharing/slow parachains:
+  // Useful for parachains, which want a 12 second block time. - With async
+  backing, all parachains will be of this type in the beginning:
+  Alternating2(Claim, Claim),
```

Comment:
Not sure I'm getting this right: the `Alternating` once should go through the parathread queue as well since we don't want to have a core wasted for when they don't produce a block, right? They would obviously jump the auction and get priority to guarantee their para block every n relay blocks?

Reply:
This is just an example for other assignment providers than parathreads: Two  parachains sharing a single core. The parachain assignment provider would always return `Assignment::List`.

But I am revisiting this as we speak. Found some issues with this design.

Reply:
Yep, that was my initial idea: Just always return a list, for parachains it is always the same entry. I am currently iterating on this, I think not even lists work well .. but let's see.

Reply:
So you are in agreement that we should relax the parachain definition to add a "block building frequency"? And not have a new assignment provider for every "block building frequency"?